hydro_lang/src/stream.rs

```diff
@@ -1573,6 +1574,120 @@ impl<'a, T, L: Location<'a> + NoTick + NoAtomic, B, Order> Stream<T, L, B, Order
         Stream::new(Atomic { tick: tick.clone() }, self.ir_node.into_inner())
     }
 
+    /// Consumes a stream of `Future<T>`, produces a new stream of the resulting `T` outputs.
+    /// Future outputs are produced as available, regardless of input arrival order.
+    ///
+    /// # Example
+    /// ```rust
+    /// # use std::collections::HashSet;
+    /// # use dfir_rs::futures::StreamExt;
+    /// # use hydro_lang::*;
+    /// # tokio_test::block_on(test_util::stream_transform_test(|process| {
+    /// let tick = process.tick();
+    /// process.source_iter(q!([2, 3, 1, 9, 6, 5, 4, 7, 8]))
+    ///     .map(q!(|x| async move {
+    ///         // tokio::time::sleep works, import then just sleep does not, unsure why
```

Comment:
Let's not put this inside the test. Also, the reason this happens is that for doctests specifically, we don't have infrastructure to extract the imports used outside the quoted code.